### ISSUE TITLE
Systeminfo starts with Error "err from getObjectList: TypeError: ..."

### ISSUE THREAD

**Issue by Sayonara-Rolo** [iobroker-community-adapters/ioBroker.systeminfo#214]
Hi ioBroker Community,

When I start "systeminfo" I get the following **Error**
"err from getObjectList: TypeError: Cannot read private member from an object whose class did not declare it"
and the **Warning**
"Unhandled error in main: TypeError: Cannot read private member from an object whose class did not declare it"
See attached Picture "systeminfo_Protocol240217.jpg"
![systeminfo_Protocol240217](https://github.com/iobroker-community-adapters/ioBroker.systeminfo/assets/160360188/3d9a3902-b78d-4e8c-ad61-13c8c8b5cee9)

An **instance** of systeminfo is created, which appears to be running (**green**)
See attached Picture "systeminfo_Instance240217.jpg"
![systeminfo_Instance240217](https://github.com/iobroker-community-adapters/ioBroker.systeminfo/assets/160360188/68b2a964-8259-4d6c-94ce-15a2bc4f6136)

Apart from the Admin Object, **no** other **Objects** are **created**.
See attached Picture "systeminfo_no-Object240217.jpg"
![systeminfo_no-Object240217](https://github.com/iobroker-community-adapters/ioBroker.systeminfo/assets/160360188/4d44314e-854d-4213-b1b3-3c930e80e0c6)

Systeminfo is up to Date with the Version 1.1.0.

What could be the Reason for this?

Thank,
Sayonara-Rolo

**Comment by xbmc-freak** [comment]
I have the same problem!
Can any one Help us?

**Comment by micki88** [comment]
After I updated to js-controller 5.0.12, this same error apeared...

**Comment by lammy000** [comment]
me 2

`<html>
<body>

systeminfo.0 | 2024-03-25 09:49:02.916 | error | err from getObjectList: TypeError: Cannot read private member from an object whose class did not declare it
-- | -- | -- | --



</body>
</html>`

**Comment by pierangelof** [comment]
Hi, 
I have the same problem. Have you found in the meanwhile a solution?

**Comment by Sayonara-Rolo** [comment]
Hi,
the problem still exists.
It seems that a number of users have the same incorrect behavior.
It's a shame that the developer didn't get in touch.
It seems to me that this adapter is no longer being maintained :(

**Comment by mcm1957** [comment]
> Hi, the problem still exists. It seems that a number of users have the same incorrect behavior. It's a shame that the developer didn't get in touch. It seems to me that this adapter is no longer being maintained :(

I would suggest that you calm down a little bit.

If you can analyze / fix the code you are very welcome to provide a PR. And you are right, there's no active maintainer for this adapter currently.

If you did not recognize - work at ioBroker is done be volunteers at their free time and without any paiment. Its easy to consume and to request only. Please help by supporteing ioBroker with your know how / coding or whtever you can provide. Or develop your own system. No developer spending its time need to be attacked by "its a shame ...". Such statements will likely reduce the number of active Developers only.

Have a nice day
mcm1957

**Comment by Sayonara-Rolo** [comment]
> > Hi, the problem still exists. It seems that a number of users have the same incorrect behavior. It's a shame that the developer didn't get in touch. It seems to me that this adapter is no longer being maintained :(
> 
> I would suggest that you calm down a little bit.
> 
> If you can analyze / fix the code you are very welcome to provide a PR. And you are right, there's no active maintainer for this adapter currently.
> 
> If you did not recognize - work at ioBroker is done be volunteers at their free time and without any paiment. Its easy to consume and to request only. Please help by supporteing ioBroker with your know how / coding or whtever you can provide. Or develop your own system. No developer spending its time need to be attacked by "its a shame ...". Such statements will likely reduce the number of active Developers only.
> 
> Have a nice day mcm1957


I don't want to or didn't mean to attack anyone here. I apologize for that.
I would also like to expressly praise and appreciate the voluntary work of the developers.
And I admire the work that some developers put into the adapters.

On the other hand, I don't think it's good for users if they no longer receive feedback from a developer.
In my opinion, this reduces confidence in the adapters.
In this case, you could leave a note saying that the adapter is no longer being maintained.

Once again, many thanks for the commitment and dedication of the developers.

Unfortunately, I don't have the know-how to program or debug the adapter myself.
Thanks, Roland

**Comment by mcm1957** [comment]
Thanks for the feedback adn explaining things.

Yes I can understand that you are frustracted. But this (nad many other) adapter lacks an active maintainer. So you nobody responded either. I cannot promise to look at it at a short time - my list is sinply tooo long. But I am aware that this adapter needs a maintainance.

**Comment by lfischer85** [comment]
> At Whever find time to look into it ... or maybe @Sayonara-Rolo wanne try it out and report back?
> 
> i think the issues is in https://github.com/iobroker-community-adapters/ioBroker.systeminfo/blob/master/myAdapter.js#L63-L68 ... all here ".bind(adapter)" is missing for all these function calls because they get patched . This woukd no longer since ages and there could be other "ex Frank Joke Adapters" where this is already fixed. I think these places could be changed like https://github.com/iobroker-community-adapters/ioBroker.radar2/blob/master/fjadapter-core.js#L325-L332 ... ideally this could already do the trick ... if there is more we see afterwards ;-)

You are right, I have replaced these lines as you suggested, and my Systeminfo Adapter is working again.
Unfortunately I don't know how to properly create a pull request, but I wanted to let you and others know, that this seems to do the trick.
Thank you for pointing this out.

**Comment by lfischer85** [comment]
After making the following changes, the adapter is now working for me and updating / creating objects again:

As mentioned by [Apollon77](https://github.com/Apollon77) replace:
https://github.com/iobroker-community-adapters/ioBroker.systeminfo/blob/master/myAdapter.js#L63-L68
with 
https://github.com/iobroker-community-adapters/ioBroker.radar2/blob/master/fjadapter-core.js#L325-L332

Additionally replace
https://github.com/iobroker-community-adapters/ioBroker.systeminfo/blob/master/myAdapter.js#L148-L149
with 
https://github.com/iobroker-community-adapters/ioBroker.radar2/blob/master/fjadapter-core.js#L333-L336

and
https://github.com/iobroker-community-adapters/ioBroker.systeminfo/blob/master/myAdapter.js#L152
with
https://github.com/iobroker-community-adapters/ioBroker.radar2/blob/master/fjadapter-core.js#L441

**Comment by printid** [comment]
Great job, thank you! The adaper works with this changes.

**Comment by mcm1957** [comment]
Please ürovide a PR to integrare the change into the official code base

**Comment by mcm1957** [comment]
@printid 
Thanks for your note. I already saw the PR created by @lfischer85. But I forgot about it and nobody reminded me :-).
I'll create a new release this week.
